src/ui/modal.tsx

```diff
@@ -102,25 +111,37 @@ const Modal = (props: ModalParams) => {
     <Overlay closeFunc={props.closeFunc}>
       <Container mode={props.mode}>
         {loaderState.type == 'OTP_ERROR' ? (
-          <OTPError action={() => dispatch('RESET')} />
+          <OTPError action={() => dispatch('RESET')} mode={props.mode} />
         ) : (
           <>
             <Header compact={props.options.compact} logo={props.logo} />
             <EmailLogin
               email={email}
               setEmail={setEmail}
               loginWithOTPStart={otpLogin}
+              mode={props.mode}
             />
             {props.loginList.length > 0 ? (
               <>
-                <Separator text="or continue with" />
+                <Separator text="or" />
                 <SocialLogin
                   loginWithSocial={socialLogin}
                   loginList={props.loginList}
                   mode={props.mode}
+                  setShowMore={() => onShowMore(true)}
```

Comment:
Why not replace `onShowMore` with `setShowMore`? What is the purpose of `onShowMore` if you are creating an arrow function anyway?